src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/SocketInput.cs

```diff
@@ -33,6 +32,8 @@ public class SocketInput : ICriticalNotifyCompletion, IDisposable
         private bool _consuming;
         private bool _disposed;
 
+        private TaskCompletionSource<bool> _tcs = new TaskCompletionSource<bool>();
```

Comment:
`TaskCompletionSource<bool>` implies `Result` is useful and might be `false`. Since it never will be, `TaskCompletionSource<object>` is probably better.